lib/powerpoint/google_services/presentation.rb

```diff
@@ -0,0 +1,194 @@
+# frozen_string_literal: true
+
+require_relative "base"
+
+module Powerpoint
+  module GoogleServices
+    class Presentation < Base
+      attr_accessor :presentation_id
+      attr_reader :template_name
+
+      METADATA_STRUCTURE = {
+        intro: [],
+        textual: [],
+        picture: [],
+        bullet: [],
+      }
+      BULLET_GLYPH = [
+        "BULLET_DISC_CIRCLE_SQUARE",
+        "BULLET_DIAMONDX_ARROW3D_SQUARE",
+        "BULLET_CHECKBOX",
+        "BULLET_ARROW_DIAMOND_DISC",
+        "BULLET_STAR_CIRCLE_SQUARE",
+        "BULLET_ARROW3D_CIRCLE_SQUARE",
+        "BULLET_LEFTTRIANGLE_DIAMOND_DISC",
+        "BULLET_DIAMONDX_HOLLOWDIAMOND_SQUARE",
+        "BULLET_DIAMOND_CIRCLE_SQUARE",
+        "NUMBERED_UPPERALPHA_ALPHA_ROMAN",
+      ]
+
+      def initialize(template_name = "")
+        super()
+
+        @template_name = template_name
+      end
+
+      def presentation(reload: false)
+        return @presentation if @presentation && !reload
+
+        create_presentation! unless presentation_id
+
+        presentation_source = slides.get_presentation(presentation_id)
+
+        @presentation = {
+          data: presentation_source,
+          metadata: map_template(presentation_source),
+        }
+      end
+
+      def duplicate_slide(object_id_prop, index)
+        new_slide_id_prop = "COPY_#{object_id_prop}#{index}"
+
+        duplicate_request = Google::Apis::SlidesV1::DuplicateObjectRequest.new(
+          object_id_prop: object_id_prop,
```

Comment:
Isso pode ser implícito

```suggestion
          object_id_prop:,
```